### ISSUE TITLE
Setting E0_DRIVER_TYPE to TMC2208 for Linear Advances breaks firmware

### ISSUE THREAD

**Issue by nullstalgia** [Jyers/Marlin#1312]
Howdy, it's been a while.

Came in to grab whatever came upstream, and ran into an issue whenever I try to use my config that I melded with the newest templates.

It would compile, but the screen would stay black when flashing, even on another SD card.

My configs:

https://gist.github.com/nullstalgia/6fa19037827acb4137735c8bd8f510fa

**Comment by tititopher68-dev** [comment]
@nullstalgia 
In your attached configuration.h file, you declared for E0 a TMC2208_Standalone instead of TMC2208.

**Comment by nullstalgia** [comment]
@tititopher68-dev 

True, but that is the only difference that allows the firmware to be loaded.

The rest of the config is accurate.

**Comment by Jyers** [comment]
@nullstalgia My config is running fine and I also use TMC2208, so I'd either point to some other weird SD issue, or try comparing your config to mine

**Comment by nullstalgia** [comment]
On a fresh clone of the JyersUI branch, I got this with Jyer's Config in the folder

![image](https://user-images.githubusercontent.com/20761757/129497411-5c756cdc-000a-45e7-b394-7b963a822476.png)

Then, after a compile, it changed to this

![image](https://user-images.githubusercontent.com/20761757/129497424-5c1b23a1-6337-435c-8548-393cba6f22fb.png)

Changing it to Standalone clears all workspace errors

![image](https://user-images.githubusercontent.com/20761757/129497435-e45272a3-a16b-4df9-b872-27da67cdbca3.png)

**Comment by Jyers** [comment]
Ahhhh, that's just a VSC issue. See #932. That will fix it

**Comment by nullstalgia** [comment]
@Jyers I've been able to narrow it down to something weird.

If I set my pin to say, PA13, It flashes properly, I get a TMC CONNECTION ERROR on the screen, and M122 errors out with ALL LOW.

Rolling back to my older firmware, running M122 returns me this at the bottom:

![image](https://user-images.githubusercontent.com/20761757/129536708-ede4d784-31a8-4ea4-bc2d-a49d78001be6.png)

This is the guide I followed for my mod:

https://drive.google.com/file/d/15cUf10lMxW4NUHE9qVAlamXmULdrOJxz/view

And even the [mod author you linked](https://github.com/Jyers/Marlin/issues/1266) used the same method (Red wire across the MCU)

![image](https://user-images.githubusercontent.com/20761757/129536832-2d96f91c-e40d-4661-8c1f-fdeb1b7a16af.png)

I get no serial output from the printer, and the screen is black the whole time.

So, help is still needed to get this working.

**Comment by TOTA42** [comment]
@nullstalgia 
The Red wire that your image goes is actually PA3, not PA13, which is correctly stated in https://drive.google.com/file/d/15cUf10lMxW4NUHE9qVAlamXmULdrOJxz/view on pages 2 and 5.  Another issue is that TMC_UART_PIN_E isn't used in the code I believe. I think it should be E0_SERIAL_TX_PIN and E0_SERIAL_RX_PIN.
If you change your 
#define TMC_UART_PIN_E PA13
to 
  #define E0_SERIAL_TX_PIN                  PA3
  #define E0_SERIAL_RX_PIN                  PA3

 I think you'll be fine.

**Comment by Jyers** [comment]
The reason my pin is set to PA13 is because that's what I wired my driver to. You need to find what UART pin you've connect the e driver to. If you start from my config and then simply change the pin, that should work

**Comment by doubleO00** [comment]
Any thoughts on why after setting E0 from TMC2208_Standalone to TMC2208 after the solder wire mod my LCD wont show, I get a blank screen, but if I change back to TMC2208_Standalone the LCD screen lights up normal.

Trying to get Lin_Advance to work on E0,  firmware compiles with both TMC2208_Standalone and  TMC2208 with no errors

When I flash the bin file that I build with TMC2208 instead of TMC2208_Standalone I lose LCD

**Comment by tiagofreire-pt** [comment]
Are you sure that you have TMC2208, instead of TMC2209?

Your board seems to be an "B" one:

https://github.com/Jyers/Marlin/discussions/707#discussioncomment-701286

**Comment by BradleyHight** [comment]
@Jyers I am working with your config on Marlin 2.0.1 and similar to those above I receive a black/blank screen, and no serial response upon trying to flash with "E0_DRIVER_TYPE" as "TMC 2208". Upon comparing with the raw file provided in "Configuration Files" my only changes are the slice thermistor ("#define TEMP_SENSOR_0 67"), and pulling TMC UART in on the STM32 pin PA3 ("#define TMC_UART_PIN_E PA3") per [Wong Sy Ming's guide](https://drive.google.com/file/d/15cUf10lMxW4NUHE9qVAlamXmULdrOJxz/view). I also see this configuration confirmed by [u/LookAtTheShinyShiny](https://www.reddit.com/user/LookAtDaShinyShiny/) on reddit; however upon talking with him, he is running v1.3.4 with stability which I will be trying later. Yet on this branch, simply changing back to "E0_DRIVER_TYPE" as "TMC 2208_STANDALONE" allows for complete flash and stepper control while not supporting the TMC UART interface. Achieving dimensional accuracy with the high flow rate Mosquito Magnum is critical for my use case and I would greatly appreciate your help on further troubleshooting or bug fixes to achieve linear advance on my E3 V2

**Comment by BradleyHight** [comment]
Can confirm successful flash and UART response with M122 via serial with your config on v1.3.4 as recommended by u/LookAtTheShinyShiny. This issue appears to be isolated to v2.0.x. Please advise upon bugfix or if you require further testing on my end.
